src/Configuration.ts

```diff
@@ -0,0 +1,491 @@
+          hidden: false,
+          normalize: true,
+          type: "string",
+        })
+        .option("temp-instrumented-directory", {
+          alias: [],
+          default: ".syntest/instrumented",
+          description: "Path to the temporary instrumented directory",
+          group: "Directory options:",
+          hidden: false,
+          normalize: true,
+          type: "string",
+        })
+    );
+  }
+
+  configureAlgorithmOptions<T>(yargs: Yargs.Argv<T>) {
+    return (
+      yargs
+        // algorithm settings
+        .option("algorithm", {
+          alias: ["a"],
+          choices: ["Random", "NSGAII", "MOSA", "DynaMOSA", "sFuzz"],
+          default: "DynaMOSA",
+          description: "Algorithm to be used by the tool.",
+          group: "Algorithm options:",
+          hidden: false,
+          type: "string",
+        })
+        .option("population-size", {
+          alias: [],
+          default: 50,
+          description: "Size of the population.",
+          group: "Algorithm options:",
+          hidden: false,
+          type: "number",
+        })
+    );
+  }
+
+  configureBudgetOptions<T>(yargs: Yargs.Argv<T>) {
+    return (
+      yargs
+        // time settings
+        .option("total-time-budget", {
+          alias: ["t"],
+          default: 3600,
+          description: "Total time budget",
+          group: "Budget options:",
+          hidden: false,
+          type: "number",
+        })
+        .option("search-time-budget", {
+          alias: [],
+          default: 3600,
+          description: "Search time budget",
+          group: "Budget options:",
+          hidden: false,
+          type: "number",
+        })
+        .option("iteration-budget", {
+          alias: ["b"],
+          default: Number.MAX_SAFE_INTEGER,
+          description: "Iteration budget",
+          group: "Budget options:",
+          hidden: false,
+          type: "number",
+        })
+        .option("evaluation-budget", {
+          alias: ["e"],
+          default: Number.MAX_SAFE_INTEGER,
+          description: "Evaluation budget",
+          group: "Budget options:",
+          hidden: false,
+          type: "number",
+        })
+    );
+  }
+
+  configureLoggingOptions<T>(yargs: Yargs.Argv<T>) {
+    return (
+      yargs
+        // logging
+        .option("console-log-level", {
+          alias: [],
+          choices: ["debug", "error", "warn", "info", "verbose", "silly"],
+          default: "debug",
+          description: "Log level of the tool",
+          group: "Logging options:",
+          hidden: false,
+          type: "string",
+        })
+        .option("log-to-file", {
+          alias: [],
+          default: ["info", "warn", "error"],
+          description: "Which levels should be logged to file",
+          group: "Logging options:",
+          hidden: false,
+          type: "array",
+        })
+    );
+  }
+
+  configurePostProcessingOptions<T>(yargs: Yargs.Argv<T>) {
```

Comment:
This is not implemented, right?

Reply:
In solidity it seemed to be used already.